pkg/model/dev_manifest.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import "github.com/okteto/okteto/pkg/log"
+
+//DevManifest represents an okteto manifest
+type DevManifest struct {
+	Name      string          `json:"name" yaml:"name"`
+	Icon      string          `json:"icon,omitempty" yaml:"icon,omitempty"`
+	Variables []DevVariables  `json:"variables,omitempty" yaml:"variables,omitempty"`
+	Deploy    DeployInfo      `json:"deploy,omitempty" yaml:"deploy,omitempty"`
+	Dev       DevManifestDevs `json:"dev,omitempty" yaml:"dev,omitempty"`
+	Destroy   []string        `json:"destroy,omitempty" yaml:"destroy,omitempty"`
+
+	// Build        map[string]BuildInfo         `json:"build,omitempty" yaml:"build,omitempty"`
+	// Test         map[string]*TestInfo         `json:"test,omitempty" yaml:"test,omitempty"`
+	// Dependencies map[string]*DependenciesInfo `json:"dependencies,omitempty" yaml:"dependencies,omitempty"`
+}
+
+type DevManifestDevs map[string]*Dev
+
+func NewDevManifest() *DevManifest {
+	return &DevManifest{
+		Variables: make([]DevVariables, 0),
+		Dev:       make(map[string]*Dev),
+		Destroy:   make([]string, 0),
+	}
+}
+
+func NewDevManifestFromDev(dev *Dev) *DevManifest {
+	devManifest := NewDevManifest()
+	name, err := ExpandEnv(dev.Name)
+	if err != nil {
+		log.Infof("could not expand dev name '%s'", dev.Name)
+		name = dev.Name
+	}
+	devManifest.Name = name
+	devManifest.Dev[name] = dev
+	return devManifest
+}
+
+//DevVariables represents a variable
```

Comment:
This comment seems a bit weak. Is this enough for future readers of the code? Maybe a bit on it's intended use. These comments are what will describe the struct in generated go doc and you don't have the code as context in that setting. I don't know that we are using go doc but we should and it's nice regardless of our godoc status.

```
// DevVariables represents the variables that are attributed to an Okteto dev manifest.
```